src/app/topbar/topbar-container.component.ts

```diff
@@ -1,24 +1,65 @@
 import { Component, HostBinding } from '@angular/core';
+import { JwtHelper } from 'angular2-jwt';
+import { AuthService } from '../services/auth.service';
 
 @Component({
   selector: 'app-topbar-container',
   template: `
     <nav class="nav-left">
-      <a *ngIf="apiTokenEnabled" class="btn btn-default">Set API Token</a>
-      <a *ngIf="loginEnabled" class="login-link">Login</a>
+      <input type="text"
+             class="form-control nav-control"
+             [value]="token"
+             (input)="validateToken($event.target.value)">
+      <button type="button"
+              *ngIf="apiTokenEnabled"
+              class="btn btn-default btn-navbar"
+              [disabled]="isSetTokenDisabled"
+              (click)="authService.setToken(token)">
+        Set API Token
+      </button>
+      <button *ngIf="loginEnabled" class="btn btn-navbar">Login</button>
     </nav>
     <nav class="nav-right">
       <a href="#" class="btn btn-secondary">curl</a>
     </nav>
   `
 })
 export class TopbarContainerComponent {
+  private error: string = 'No Token';
+  private jwtHelper: JwtHelper = new JwtHelper();
+  private token: string = '';
 
-  constructor() { }
+  constructor(private authService: AuthService) { }
 
   @HostBinding('class')
   private classNames: string = 'navbar';
 
-  private apiTokenEnabled: boolean = false;
+  private apiTokenEnabled: boolean = true;
```

Comment:
You're probably running into:
https://github.com/angular/angular/issues/11978

When you use AOT, public/private properties are enforced at compile time, and since the template is external to the component class, any component property used in the template must be public. I believe this was a side effect of a change in the way Angular 4 compiles components/templates, but I can't find the reference for that ATM.